packages/core/src/indexing/postgres/__tests__/postgres-index-api.test.ts

```diff
@@ -59,25 +46,47 @@ describe('init', () => {
       const modelsToIndex = [StreamID.fromString(STREAM_ID_A)]
       const indexApi = new PostgresIndexApi(dbConnection, modelsToIndex, true)
       await indexApi.init()
-      const created = await listMidTables()
+      const created = await listMidTables(dbConnection)
+      const tableNames = modelsToIndex.map(asTableName)
+      expect(created).toEqual(tableNames)
+    })
+
+    test('create new table from scratch but fail table validation', async () => {
+      const INVALID_TABLE_STRUCTURE = {
+        stream_id: {
+          type: 'character varying',
+          maxLength: 254,
+          nullable: false,
+          defaultValue: null,
+        },
+      }
+
+      const modelsToIndex = [StreamID.fromString(STREAM_ID_A)]
+      const indexApi = new PostgresIndexApi(dbConnection, modelsToIndex, true)
+      await indexApi.init()
+      const created = await listMidTables(dbConnection)
       const tableNames = modelsToIndex.map(asTableName)
       expect(created).toEqual(tableNames)
+      // TODO: error check is not caught correctly... why jest... why u no workie
```

Comment:
do you plan on fixing this todo as part of this PR?

Reply:
I do, yes. Started looking into it yesterday.